src/js/components/List/index.d.ts

```diff
@@ -0,0 +1,30 @@
+import * as React from "react";
+import { A11yTitleType, AlignSelfType, GridAreaType, MarginType } from "../../utils";
+
+type SizeType = "xxsmall" | "xsmall" | "small" | "medium" | "large" | "xlarge" | string ;
+type SideType = "top" | "left" | "bottom" | "right" | "horizontal" | "vertical" | "all" ;
+type PadSizeType = "none" | "xxsmall" | "xsmall" | "small" | "medium" | "large" | "xlarge" | string ;
+type PadType = PadSizeType | {bottom?: PadSizeType,horizontal?: PadSizeType,left?: PadSizeType,right?: PadSizeType,top?: PadSizeType,vertical?: PadSizeType};
+type BorderType = boolean | SideType | {color?: string | {dark?: string,light?: string},side?: SideType,size?: SizeType};
+
+export interface ListProps {
+  a11yTitle?: A11yTitleType;
+  alignSelf?: AlignSelfType;
+  as?: string;
+  background?: string | string[];
+  border?: BorderType;
+  data?: string[] | {}[];
+  gridArea?: GridAreaType;
+  itemProps?: { [_:string]: { background?: string, border?: BorderType, pad?: PadType}};
+  margin?: MarginType;
+  onMore?: ((...args: any[]) => any);
+  onClickItem?: ((...args: any[]) => any);
+  pad?: PadType;
+  primaryKey?: string | ((...args: any[]) => any);
+  secondaryKey?: string | ((...args: any[]) => any);
```

Comment:
we are trying to get away from the usage of `any` for functions.

Reply:
ok